distribution/integration/test-common/integration-test-utils/src/main/java/org/wso2/carbon/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,485 @@
+        return System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_OS_NAME).toLowerCase();
+    }
+
+    /**
+     * provides carbon home after extracting the pack.
+     *
+     * @param context - AutomationContext
+     * @return - carbon home
+     * @throws CarbonToolsIntegrationTestException - Error while setup carbon home from carbon zip file
+     */
+    public static String getCarbonHome(AutomationContext context)
+            throws CarbonToolsIntegrationTestException {
+        try {
+            if (carbonHomePath != null) {
+                return carbonHomePath;
+            }
+
+            String carbonZip = System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_CARBON_ZIP_LOCATION);
+            CarbonServerManager carbonServerManager = new CarbonServerManager(context);
+            carbonHomePath = carbonServerManager.setUpCarbonHome(carbonZip);
+            return carbonHomePath;
+
+        } catch (IOException ex) {
+            log.error("Extracting the pack and getting the carbon home failed", ex);
+            throw new CarbonToolsIntegrationTestException("Extracting the pack and getting the " +
+                                                          "carbon home failed", ex);
+        }
+    }
+
+    /**
+     * This method is to shutdown carbon server
+     *
+     * @param portOffset - port offset
+     * @throws CarbonToolsIntegrationTestException - Error occurred while shutdown the server
+     */
+    public static void serverShutdown(int portOffset) throws CarbonToolsIntegrationTestException {
+        long time = System.currentTimeMillis() + CarbonIntegrationConstants.DEFAULT_WAIT_MS;
+        log.info("Shutting down server..");
+        boolean isLogoutSuccess = false;
+
+        try {
+            AutomationContext automationContext = new AutomationContext();
+            if (ClientConnectionUtil.isPortOpen(
+                    Integer.parseInt(ExtensionConstants.SERVER_DEFAULT_HTTPS_PORT))) {
+
+                int httpsPort = Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset;
+                String url = automationContext.getContextUrls().getBackEndUrl();
+                String backendURL = url.replaceAll("(:\\d+)", ":" + httpsPort);
+
+                ServerAdminClient serverAdminServiceClient =
+                        new ServerAdminClient(
+                                backendURL,
+                                automationContext.getSuperTenant().getTenantAdmin().getUserName(),
+                                automationContext.getSuperTenant().getTenantAdmin().getPassword());
+
+                serverAdminServiceClient.shutdown();
+                while (System.currentTimeMillis() < time && !isLogoutSuccess) {
+                    isLogoutSuccess = isServerDown(portOffset);
+                    // wait until server shutdown is completed
+                }
+                log.info("Server stopped successfully...");
+            }
```

Comment:
serverAdminServiceClient.shutdown() also throws Exception, which is also problem with test framework